generators/app/templates/src/app/components/Routes/components/RouteItem.tsx

```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import { Route, Redirect, RouteComponentProps, RouteProps } from 'react-router-dom';
+
+interface Props extends RouteProps {
+  component: React.ComponentType<RouteComponentProps<any>> | React.ComponentType<any>;
+  redirectTo?: string;
+}
+
+function RouteItem({ redirectTo, ...config }: Props) {
```

Comment:
What happens with public/private routes? Are you delegating the reddirection responsability somewhere? 
What I mean is, for example, what if a non-logged user tries to access a private route? We used to handle that kind of cases inside the route component, how do you propose we do that now? 
I think the best way would be for this to be included in the component, or at least create a HOC/wrapper to handle these changes.

Reply:
redirectTo is a function which receives a user and returns a path to redirect in case that user doesn't meet the condition to show that route. This way we could not only avoid logged users to go to the login or non-logged users to go to the home, we can check access levels or anything we'd like from the user. And with small changes (receiving something other than a user) we could also have access to some routes depending on other factors that aren't the user if we wanted to. This component now is more flexible than the previous AuthenticatedRoute.